client/state/jetpack/credentials/test/reducer.js

```diff
@@ -0,0 +1,46 @@
+import { markCredentialsAsInvalid, markCredentialsAsValid, testCredentials } from '../actions';
+import { testRequestStatus } from '../reducer';
+
+describe( 'reducer', () => {
+	describe( 'testRequestStatus', () => {
+		test( 'should return the initial state', () => {
+			expect( testRequestStatus( undefined, {} ) ).toEqual( {} );
+		} );
+
+		test( 'state should be valid if the credentials are okay', () => {
+			const stateIn = undefined;
+			const action = markCredentialsAsValid( 123000, 'main' );
+
+			const stateOut = testRequestStatus( stateIn, action );
+			expect( stateOut ).toEqual( {
+				123000: {
+					main: 'valid',
+				},
+			} );
+		} );
+
+		test( 'state should be invalid if the credentials are not okay', () => {
+			const stateIn = undefined;
```

Comment:
I see we always test with `undefined` starting state. Should we also add some tests that cover how the reducer transitions from one non-undefined state to another?

Reply:
Good one, working on it.

Reply:
Fixed and added tests for reducer transitions!